ui/app/adapters/kv/metadata.js

```diff
@@ -3,39 +3,53 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import ApplicationAdapter from '../application';
 import { encodePath } from 'vault/utils/path-encoding-helpers';
-import { kvId } from 'vault/utils/kv-id';
+import ApplicationAdapter from '../application';
+import { kvMetadataPath } from 'vault/utils/kv-path';
 
 export default class KvMetadataAdapter extends ApplicationAdapter {
   namespace = 'v1';
 
-  _urlForMetadata(backend, path) {
-    return `${this.buildURL()}/${encodePath(backend)}/metadata/${encodePath(path)}`;
+  _url(fullPath) {
+    return `${this.buildURL()}/${fullPath}`;
   }
 
   createRecord(store, type, snapshot) {
     const { backend, path } = snapshot.record;
-    const url = this._urlForMetadata(backend, path);
-
-    return this.ajax(url, 'POST', { data: this.serialize(snapshot) }).then((resp) => {
-      resp.id = kvId(backend, path, 'metadata');
-      return resp;
+    const id = kvMetadataPath(backend, path);
+    const url = this._url(id);
+    const data = this.serialize(snapshot);
+    return this.ajax(url, 'POST', { data }).then(() => {
+      return {
+        id,
+        data,
+      };
     });
   }
 
-  queryRecord(store, type, query) {
-    const { path, backend } = query;
-    return this.ajax(this._urlForMetadata(backend, path), 'GET').then((resp) => {
-      resp.id = kvId(backend, path, 'metadata');
+  // TODO: replace this with raw request for metadata request?
+  query(store, type, query) {
+    const { backend } = query;
+    return this.ajax(this._url(`${encodePath(backend)}/metadata?list=1`), 'GET').then((resp) => {
```

Comment:
No, the response here is the list, so `resp.data.keys` gets hydrated into the list of secrets. But we need to add `backend` to they payload so that the ID's can be properly hydrated in the serializer